gcs/src/main/scala/blobstore/gcs/GcsPath.scala

```diff
@@ -0,0 +1,38 @@
+package blobstore
+package gcs
+
+import java.time.Instant
+
+import cats.data.Chain
+import com.google.cloud.storage.BlobInfo
+
+case class GcsPath(blobInfo: BlobInfo) extends Path {
```

Comment:
Should we expose this as a `val` to keep this path consistent with the S3Path (which as `val s3entity: val s3Entity: Either[S3Path.S3Dir, S3Path.S3File]`)? This would enable storage class filtering on GCS as well

Reply:
`GcsPath` is a case class so it is possible to access `BlobInfo` and filter on storage class.

Reply:
Regarding `S3Path`, it might change in #40